data-prepper-plugins/blocking-buffer/src/test/java/com/amazon/dataprepper/plugins/buffer/blockingbuffer/BlockingBufferTests.java

```diff
@@ -113,7 +139,24 @@ public void testWriteIntoEmptySpaceAfterCheckedRead() throws TimeoutException {
         blockingBuffer.checkpoint(readResult.getValue());
 
         // Then
-        blockingBuffer.write(new Record<>("TIMEOUT"), TEST_WRITE_TIMEOUT);
+        blockingBuffer.write(new Record<>("REFILL_THE_BUFFER"), TEST_WRITE_TIMEOUT);
+    }
+
+    @Test
+    public void testWriteAllIntoEmptySpaceAfterCheckedRead() throws Exception {
+        // Given
+        final BlockingBuffer<Record<String>> blockingBuffer = new BlockingBuffer<>(2, TEST_BATCH_SIZE,
+                TEST_PIPELINE_NAME);
+        assertThat(blockingBuffer, notNullValue());
+        final Collection<Record<String>> testRecords = generateBatchRecords(2);
+        blockingBuffer.writeAll(testRecords, TEST_WRITE_TIMEOUT);
+
+        // When
+        final Map.Entry<Collection<Record<String>>, CheckpointState> readResult = blockingBuffer.read(TEST_BATCH_READ_TIMEOUT);
+        blockingBuffer.checkpoint(readResult.getValue());
+
+        // Then
+        blockingBuffer.writeAll(testRecords, TEST_WRITE_TIMEOUT);
```

Comment:
This test should probably read the buffer after this to verify the data is present.